server/testsupport/ciagInductionListResponseTestDataBuilder.ts

```diff
@@ -1,9 +1,9 @@
-import type { CiagInduction, CiagInductionListResponse } from 'ciagInductionApiClient'
+import type { CiagInductionSummaryListResponse, CiagInductionSummaryResponse } from 'educationAndWorkPlanApiClient'
 import { aLongQuestionSetCiagInduction } from './ciagInductionTestDataBuilder'
 
 export default function aValidCiagInductionListResponse(options?: {
-  ciagProfileList?: CiagInduction[]
-}): CiagInductionListResponse {
+  ciagProfileList?: CiagInductionSummaryResponse[]
```

Comment:
This change makes sense, but I don't understand how it was an array of `CiagInduction` before?! 🤯

Reply:
I'll try to explain ....

Previously the class types were from the CIAG API's imported swagger spec (so they were not the PLP classes, not even the PLP "versions" of the CIAG classes)
The "outer" object was a `CiagInductionListResponse` which had a single field `ciagProfileList` which was an array of `CiagInduction` (the full object !!)
This was based on the original implementation where we were calling the original CIAG API

At some point in time we implemented the "CIAG API" but in the PLP API. The paths and data types were identical, but we chose to give the classes different (better) names.
Our equivalent classes are `CiagInductionSummaryListResponse` and `CiagInductionSummaryResponse`.
What we didn't do at the time, which is what this change is about, is changing the UI codebase to use the new types from the PLP swagger spec. We changed the API endpoint, so it was calling the PLP API, but we didnt change these types here.

Because the types were identical (more on that in a sec) apart from name, it just worked. Perhaps more by luck than judgement, but it worked.

Now .... are the class types identical? 😁 🤔  I think you know the answer to that is no! 🤣 which I guess is perhaps where your question has come from ...

Our "outer wrapper" class `CiagInductionSummaryListResponse` is identical to the CIAG API class `CiagInductionListResponse` because they both contain a single field `ciagProfileList` which contains an array of `CiagInduction` (CIAG API class) / `CiagInductionSummaryResponse` (PLP API class)

Are our `CiagInductionSummaryResponse` PLP API class and CIAG API class `CiagInduction` identical ?
No ...  our `CiagInductionSummaryResponse` is (as the name suggests) a summary, rather than the full induction. But it contains a subset of the fields from `CiagInduction`, crucially the "important" ones that the UI reads.

Does that make sense?

Reply:
Ah .... I misunderstood your question ..... it was about the test data builder, and whether they were building the right thing, given that the PLP's summary object is indeed a summary, rather than a full induction!
Gotcha! Just making that change now 👍